accio-tests/src/test/java/io/accio/testing/TestConfigResource.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.accio.testing;
+
+import com.google.common.collect.ImmutableMap;
+import io.accio.base.dto.Manifest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static io.accio.base.config.AccioConfig.ACCIO_DATASOURCE_TYPE;
+import static io.accio.base.config.AccioConfig.ACCIO_DIRECTORY;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_CRENDITALS_KEY;
+import static io.accio.base.config.ConfigManager.ConfigEntry.configEntry;
+import static io.accio.base.dto.Manifest.MANIFEST_JSON_CODEC;
+import static io.accio.testing.AbstractTestFramework.withDefaultCatalogSchema;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+public class TestConfigResource
+        extends RequireAccioServer
+{
+    private Path mdlDir;
+
+    @Override
+    protected TestingAccioServer createAccioServer()
+    {
+        Manifest manifest = withDefaultCatalogSchema()
+                .build();
+
+        try {
+            mdlDir = Files.createTempDirectory("acciomdls");
+            Path accioMDLFilePath = mdlDir.resolve("acciomdl.json");
+            Files.write(accioMDLFilePath, MANIFEST_JSON_CODEC.toJsonBytes(manifest));
+        }
+        catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
+
+        ImmutableMap.Builder<String, String> properties = ImmutableMap.<String, String>builder()
+                .put(ACCIO_DIRECTORY, mdlDir.toAbsolutePath().toString())
+                .put(ACCIO_DATASOURCE_TYPE, "duckdb");
```

Comment:
We can use `DataSourceType.DUCKDB.name()` instead of the string to reduce string. WDYT?

Reply:
hmm. I think we can keep this to assert this configuration is case insensitive.

Reply:
Case insensitive should be other test case. Not append with this case.